embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/SpansServiceImpl.kt

```diff
@@ -160,11 +160,12 @@ internal class SpansServiceImpl(
 
     override fun flushSpans(appTerminationCause: EmbraceAttributes.AppTerminationCause?): List<EmbraceSpanData> {
         synchronized(completedSpans) {
+
+            val telemetryAttributes = embraceTelemetryService.getTelemetryAttributes()
```

Comment:
Right not, session spans don't survive native crashes and sudden process terminations, so this would not be recorded in those cases. And even when we address that, it'll likely go through a different code path that won't access to this.

I think this is OK for now, but I would add a comment here about this limitation and how we need to address this in the future when we address ending in-progress spans left dangling after a crash

Reply:
We could cache telemetry on its own? Either way, yeah, let's think about it later.